File: projects/code_combat/7_Sarven_Desert/291-Sarven_Sum/sarvenSum.js
// To disable the fire-traps add the lowest trap.value to the highest value.
// Move to the white X and say the answer to Kitty the cougar.
// Defeat all the ogres if you dare.
// Once all ogres are defeated move to the red X.
// Look out for potions to boost your health.

function onSpawn() {
    var item = hero.findNearestItem();
    if (item)
        pet.fetch(item);
}

var whiteX = {x:27, y:42};
var redX = {x:151 , y: 118};
pet.on("spawn", onSpawn);

var traps = hero.findHazards();
var max = 0;
var min = 9999;

for (var i = 0; i < traps.length; i++) {
    var value = traps[i].value;
    if (value) {
        if (value < min)
            min = value;
        if (value > max)
            max = value;
    }
}

hero.moveXY(whiteX.x, whiteX.y);
hero.say(min + max);
hero.moveXY(redX);

File: projects/code_combat/7_Sarven_Desert/291-Sarven_Sum/sarven_sum.py
# You need a high-level boots to access a "move" action
# You need the Emperor's gloves to cast "Chain Lightning"
# You need the Unholy Tome IV+ to cast strong enough "Drain Life"

def onSpawn():
    item = hero.findNearestItem()
    if item:
        pet.fetch(item)


whiteX = {x:27, y:42}
redX = {x:151 , y: 118}
pet.on("spawn", onSpawn)

traps = hero.findHazards()
max = 0
min = 9999

for trap in traps:
    value = trap.value
    if value:
        if value < min:
            min = value
        if value > max:
            max = value

hero.moveXY(whiteX.x, whiteX.y)
hero.say(min + max)

while True:
    enemy = hero.findNearestEnemy()
    if enemy:
        if hero.canCast("chain-lightning"):
            hero.cast("chain-lightning", enemy)
        else:
            hero.cast("drain-life", enemy)
    hero.move(redX);
